slack-plugin/src/main/kotlin/slack/gradle/artifacts/SgpArtifacts.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.gradle.artifacts
+
+import org.gradle.api.Named
+import org.gradle.api.attributes.Attribute
+
+internal interface SgpArtifacts : Named {
+  companion object {
+    @JvmField
+    val SGP_ARTIFACTS_ATTRIBUTE: Attribute<SgpArtifacts> =
+      Attribute.of("sgp.internal.artifacts", SgpArtifacts::class.java)
+  }
+
+  enum class Kind(
+    val declarableName: String,
+    val artifactName: String,
+  ) {
+    SKIPPY_UNIT_TESTS("skippyUnitTests", "skippy-unit-tests"),
+    SKIPPY_LINT("skippyLint", "skippy-lint"),
+    SKIPPY_AVOIDED_TASKS("skippyAvoidedTasks", "skippy-avoided-tasks"),
+    SKIPPY_ANDROID_TEST_PROJECT("skippyAndroidTestProject", "skippy-android-test-project"),
+    SKIPPY_DETEKT("skippyDetekt", "skippy-detekt"),
+    ANDROID_TEST_APK_DIRS("androidTestApkDirs", "android-test-apk-dirs"),
+    DAGP_MISSING_IDENTIFIERS("dagpMissingIdentifiers", "dagp-missing-identifiers"),
+    MOD_STATS_STATS_FILES("modStatsFiles", "mod-stats-files"),
+  }
```

Comment:
if I were to try to turn this into a library, what are your thoughts on how I'd handle this sort of thing? My project and your project have different artifact kinds, but the `SgpArtifacts` type is baked into the resolver/publisher APIs.

Reply:
I updated the PR with a sketch of a more agnostic API, but it forgoes use of `Named` and instead expects the artifact you give it to implement equals(). This takes a page out of AGP's artifact handling (i.e. like `SingleArtifact`'s subtypes). I think this works pretty well